### PR TITLE
Feature - add IoT Hub dependency tracking

### LINKED ISSUE #54: Provide capability to track an Azure IoT Hub dependency
**Is your feature request related to a problem? Please describe.**
Provide capability to track an Azure IoT Hub dependency.

**Describe the solution you'd like**
We should provide a `LogIoTHubDependency` overload for `ILogger` which is optimized for Azure IoT Hub and correctly generates the [Azure Application Insights App Map](https://docs.microsoft.com/en-gb/azure/azure-monitor/app/app-map).

**Describe alternatives you've considered**
None

**Additional context**
Dependency type should be used from [official SDK](https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/WEB/Src/DependencyCollector/DependencyCollector/Implementation/RemoteDependencyConstants.cs).

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #54

### CODE REVIEW

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs [hunk 1]
```diff
@@ -384,6 +384,44 @@ public static void LogEventHubsDependency(this ILogger logger, string namespaceN
...
+        /// <param name="duration">Duration of the operation</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogIotHubDependency(this ILogger logger, string namespaceName, string iotHubName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger));
+            Guard.NotNullOrWhitespace(iotHubName, nameof(iotHubName));
+
+            context = context ?? new Dictionary<string, object>();
+
+            logger.LogInformation(DependencyFormat, "Azure Iot Hub", namespaceName, iotHubName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:422]:
```suggestion
            logger.LogInformation(DependencyFormat, "Azure IoT Hub", namespaceName, iotHubName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
```

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs [hunk 2]
```diff
@@ -384,6 +384,44 @@ public static void LogEventHubsDependency(this ILogger logger, string namespaceN
...
+        ///     Logs an Azure Iot Hub Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="namespaceName">Namespace of the resource</param>
+        /// <param name="iotHubName">Name of the Event Hub resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="startTime">Point in time when the interaction with the dependency was started</param>
+        /// <param name="duration">Duration of the operation</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogIotHubDependency(this ILogger logger, string namespaceName, string iotHubName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:415]: Azure IoT Hub is a per-instance service and doesn't have notion of a namespace
```suggestion
        public static void LogIotHubDependency(this ILogger logger, string iotHubName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
```

File: src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs [hunk 3]
```diff
@@ -384,6 +384,44 @@ public static void LogEventHubsDependency(this ILogger logger, string namespaceN
...
+        /// <summary>
+        ///     Logs an Azure Iot Hub Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="namespaceName">Namespace of the resource</param>
+        /// <param name="iotHubName">Name of the IoT Hub resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="measurement">Measuring the latency to call the dependency</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogIotHubDependency(this ILogger logger, string namespaceName, string iotHubName, bool isSuccessful, DependencyMeasurement measurement, Dictionary<string, object> context = null)
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:396]: Azure IoT Hub is a per-instance service and doesn't have notion of a namespace

```suggestion
        public static void LogIotHubDependency(this ILogger logger, string iotHubName, bool isSuccessful, DependencyMeasurement measurement, Dictionary<string, object> context = null)
```

Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:396]: Should we than use something else for dependency data?

Comment (tomkerkhove) [reply src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs:396]: No, I think just the IoT Hub name is good for now.

If something else makes sense than we can still add it.

**Pull request merged**